go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -566,7 +566,13 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 	}
 
 	// CNI depends on the flows from port security, delay setting it until end
-	lsp.PortSecurity = addresses
+	if !bnc.DisablePortSecurity() {
+		lsp.PortSecurity = append(lsp.PortSecurity, lsp.Addresses...)
+	}
+	if bnc.AllowL2Unknown() {
+		const allowUnknownAddresses = "unknown"
+		lsp.Addresses[0] = fmt.Sprintf("%s %s", lsp.Addresses[0], allowUnknownAddresses)
```

Comment:
Are we sure this shouldn't be like:

lsp.Addresses[0] = "\<mac> \<ip>"
lsp.Addresses[1] = "unknown"

That's how it looks like it should be reading the ovn doc.

Reply:
I have the slight recollection of trying to do that and it didn't work. 

I'll check again, and will try to come up w/ a way to have a proper e2e test for this. 

For now, what I can do is try what you say, and expect the e2e test to fail (assuming I am right ...).

Reply:
Maybe check out how it ends up being set in the db with ovn-nbctl

Reply:
Otherwise I would check with @dceara to make sure we have the correct interpretation of what we are doing.

Reply:
Seems you were right. Changed it accordingly.